File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        result = 0
        l=0
        r=1

        for r in range(len(prices)):
            profit = 0;
            if prices[r] > prices[l]:
                profit = prices[r] - prices[l];
            else:
                l=r
            result = max(result,profit);
        
        return result;
        

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-8.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l=0;
        int result = 0;
        Set<Character> set = new HashSet<>();
        for(int r=0; r< s.length(); r++){
            char cr = s.charAt(r);
            char cl = s.charAt(l);
            while(set.contains(cr)){
                l++;
                set.remove(cl);
                cl = s.charAt(l);
            }
            set.add(cr);
            result = Math.max(result,(r-l+1));
        }
        return result;
    }
}
